src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -113,6 +125,38 @@ public OpenApiDocument GetSwagger(string documentName, string host = null, strin
             return (applicableApiDescriptions, swaggerDoc, schemaRepository);
         }
 
+        private async Task<(IEnumerable<ApiDescription>, OpenApiDocument, SchemaRepository)> GetSwaggerDocumentWithoutDocumentFiltersAsync(string documentName, string host = null, string basePath = null)
```

Comment:
This seems to introduce a lot of duplication between the two implementations. Can it be refactored to just make the existing methods async and then iterate over both sets?

Reply:
I was thinking about something like that, but I am not sure how I can call these async methods in the existing non-async code-path?

Don't we need to support `ISwaggerProvider.GetSwagger()` still?

Reply:
I only looked at the diff, I didn't go digging through all the code. Can you point me at the bit where you're looking?

We would need to keep the existing methods/support though, yes.

Reply:
I figured out a way to remove the duplication.